tutorials/iot/graphical-snaps/graphical-snaps-xwayland.md

```diff
@@ -0,0 +1,341 @@
+
+Set WAYLAND_SOCKET_DIR to "$SNAP_DATA/wayland":
+```yaml
+...
+     WAYLAND_SOCKET_DIR: $SNAP_DATA/wayland
+...
+```
+
+Add a `plugs` stanza:
+```yaml
+...
+plugs:
+   wayland-socket-dir:
+    content: wayland-socket-dir
+    interface: content
+    target: $SNAP_DATA/wayland
+    default-provider: mir-kiosk
+...
+```
+   
+This additional snippet causes snapd to bind-mount the mir-kiosk Wayland socket directory into the glmark2-wayland’s namespace, in a location of its desire: $SNAP_DATA/wayland. Then we update the WAYLAND_SOCKET_DIR variable to match.
+
+### The full YAML file:
+
+```yaml
+name: iot-example-graphical-xwayland-snap
+version: 0.1
+summary: glxgears on XWayland
+description: |
+   glxgears on XWayland
+confinement: strict
+grade: devel
+
+apps:
+ glxgears-kiosk:
+   command: xwayland-kiosk-launch glxgears
+   environment:
+     XWAYLAND_FULLSCREEN_WINDOW_HINT: title="glxgears"
+     WAYLAND_SOCKET_DIR: $SNAP_DATA/wayland
+   plugs:
+     - opengl
+     - wayland
+
+parts:
+ glxgears:
+   plugin: nil
+   after: [ xwayland-kiosk-helper ]
+   stage-packages:
+     - mesa-utils
+
+plugs:
+   wayland-socket-dir:
+    content: wayland-socket-dir
+    interface: content
+    target: $SNAP_DATA/wayland
+    default-provider: mir-kiosk
+```
+
+Check this builds locally:
+```bash
+snapcraft cleanbuild
+```
+
+## Building for different architectures
+duration: 10:00
+
+One day, perhaps, snapcraft will fully support cross building with the `--target-arch` option. But getting that to work is beyond the scope of this tutorial. Instead we'll make use of Launchpad's builders to build the snap for all architectures (including the one your device provides).
+
+Create a github repository for your snap and push your changes to the snap project there:
+
+```bash
+git commit -a
+git remote remove origin
+git remote add origin https://github.com/<project>/<repo>.git
+git push -u origin master
+```
+
+Now [setup your build on Launchpad](https://docs.snapcraft.io/build-snaps/ci-integration). *Note that you will need to use the same snap name in the store as in name in `name:`, so chose something unique to make your life easier.*
+
+Don't bother with publishing to the store (yet) you can download the snap and deploy it as follows:
+
+On your desktop go to the snap webpage (e.g. https://code.launchpad.net/~alan-griffiths/+snap/iot-example-graphical-xwayland-snap), find the build for your device architecture and download it and copy to your device. For example:
+```bash
+wget https://code.launchpad.net/~alan-griffiths/+snap/iot-example-graphical-xwayland-snap/+build/228782/+files/iot-example-graphical-xwayland-snap_0.1_arm64.snap
+scp iot-example-graphical-xwayland-snap_0.1_arm64.snap alan-griffiths@192.168.1.159:~
+```
+On your ssh session to your device:
+```bash
+snap install --dangerous iot-example-graphical-xwayland-snap_0.1_arm64.snap --devmode
+snap connect iot-example-graphical-xwayland-snap:wayland-socket-dir mir-kiosk:wayland-socket-dir
```

Comment:
Explain what happens here? I believe we plan for autoconnection, but @gerboland is fighting himself on how to support the `XDG_RUNTIME_DIR` situation (that wayland hardcodes as its main socket directory).